scrapy/mail.py

```diff
@@ -82,12 +70,37 @@ def send(self, to, subject, body, cc=None, attachs=(), mimetype='text/plain', ch
                 part = MIMEBase(*mimetype.split('/'))
                 part.set_payload(f.read())
                 Encoders.encode_base64(part)
-                part.add_header('Content-Disposition', 'attachment; filename="%s"' \
-                    % attach_name)
+                part.add_header(
+                    'Content-Disposition', 'attachment; filename="%s"' % attach_name)
                 msg.attach(part)
         else:
             msg.set_payload(body)
 
+        return msg
+
+
+class MailSender(BaseMailSender):
+
+    def __init__(self, smtphost='localhost', mailfrom='scrapy@localhost',
+            smtpuser=None, smtppass=None, smtpport=25, smtptls=False, smtpssl=False, debug=False):
+        self.smtphost = smtphost
+        self.smtpport = smtpport
+        self.smtpuser = _to_bytes_or_none(smtpuser)
+        self.smtppass = _to_bytes_or_none(smtppass)
+        self.smtptls = smtptls
+        self.smtpssl = smtpssl
+        self.mailfrom = mailfrom
+        self.debug = debug
+
+    @classmethod
+    def from_settings(cls, settings):
+        return cls(settings['MAIL_HOST'], settings['MAIL_FROM'], settings['MAIL_USER'],
+            settings['MAIL_PASS'], settings.getint('MAIL_PORT'),
+            settings.getbool('MAIL_TLS'), settings.getbool('MAIL_SSL'))
```

Comment:
I know it was like this before, but with such long lists of arguments it usually makes more sense to use kwargs - much less chance to pass a value to a wrong argument.